include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -177,8 +177,27 @@ class binary_writer
 
             case value_t::number_float:
             {
-                oa->write_character(get_cbor_float_prefix(j.m_value.number_float));
-                write_number(j.m_value.number_float);
+		const auto valueD = j.m_value.number_float;
+		const float valueF = static_cast<float>(valueD);
+		if (std::isnan(valueD)) {
```

Comment:
Please include `<cmath>` to include/nlohmann/detail/output/binary_writer.hpp to use `std::isnan` and `std::isinf`.